dataherald/sql_database/models/types.py

```diff
@@ -93,16 +108,25 @@ def validate_uri(cls, input_string):
         match = re.match(pattern, input_string)
         if not match:
             raise InvalidURIFormatError(f"Invalid URI format: {input_string}")
+        return match.group(1)
+
+    @classmethod
+    def set_dialect(cls, input_string):
+        for dialect in SupportedDialects:
+            if dialect.value in input_string:
+                return dialect.value
+        return None
 
     @validator("connection_uri", pre=True, always=True)
-    def connection_uri_format(cls, value: str):
+    def connection_uri_format(cls, value: str, values):
```

Comment:
what's the difference between `value` and `values`?

Reply:
value is the value for connection_uri, values is the object itself